crates/bevy_render/src/view/window/screenshot.rs

```diff
@@ -21,51 +18,80 @@ use crate::{
     },
     renderer::RenderDevice,
     texture::TextureFormatPixelInfo,
-    RenderApp,
+    ExtractSchedule, MainWorld, Render, RenderApp, RenderSet,
+};
+use bevy_app::{First, Plugin, Update};
+use bevy_asset::{load_internal_asset, Handle};
+use bevy_derive::{Deref, DerefMut};
+use bevy_ecs::{entity::EntityHashMap, prelude::*};
+use bevy_hierarchy::DespawnRecursiveExt;
+use bevy_tasks::AsyncComputeTaskPool;
+use bevy_utils::default;
+use bevy_utils::tracing::{error, info, info_span, warn};
+use bevy_window::{PrimaryWindow, WindowRef};
+use std::ops::Deref;
+use std::sync::mpsc::{Receiver, Sender};
+use std::sync::{Arc, Mutex};
+use std::{borrow::Cow, path::Path};
+use wgpu::{
+    CommandEncoder, Extent3d, ImageDataLayout, TextureFormat, COPY_BYTES_PER_ROW_ALIGNMENT,
 };
 
-use super::ExtractedWindows;
+#[derive(Event, Deref, DerefMut)]
```

Comment:
I see this wasn't the case before but maybe now is a good chance to derive Reflect on all of these? :)

Reply:
Okay, yeah, I'm never really sure when to derive reflect, will minimally put it on the event and screenshot component?